agent/health_endpoint.go

```diff
@@ -186,6 +187,20 @@ func (s *HTTPServer) healthServiceNodes(resp http.ResponseWriter, req *http.Requ
 		prefix = "/v1/health/connect/"
 	}
 
+	// Check for ingress request only when requesting connect services
+	if connect {
+		ingress, err := getBoolQueryParam(params, "ingress")
+		if err != nil {
+			resp.WriteHeader(http.StatusBadRequest)
+			fmt.Fprint(resp, "Invalid value for ?ingress")
+			return nil, nil
+		}
+		if ingress {
+			args.Connect = false
+			args.Ingress = true
```

Comment:
This is a bit awkward, comes down to how we use these values in the RPC call...

Reply:
We could also not set `args.Connect` initially and either set it or `args.Ingress` to true here. This is really minor though.

Reply:
Like that better, did it!